fuzz/fuzz_targets/cranelift-fuzzgen.rs

```diff
@@ -51,16 +53,30 @@ fn run_in_host(compiled_fn: &CompiledFunction, args: &[DataValue]) -> RunResult
     RunResult::Success(res)
 }
 
-fuzz_target!(|testcase: TestCase| {
-    let build_interpreter = || {
-        let mut env = FunctionStore::default();
-        env.add(testcase.func.name.to_string(), &testcase.func);
+fn build_interpreter(testcase: &TestCase) -> Interpreter {
+    let mut env = FunctionStore::default();
+    env.add(testcase.func.name.to_string(), &testcase.func);
 
-        let state = InterpreterState::default().with_function_store(env);
-        let interpreter = Interpreter::new(state).with_fuel(Some(INTERPRETER_FUEL));
-        interpreter
-    };
+    let state = InterpreterState::default()
+        .with_function_store(env)
+        .with_libcall_handler(&|libcall: LibCall, args: SmallVec<[DataValue; 1]>| {
+            use LibCall::*;
+            Ok(smallvec![match (libcall, &args[..]) {
+                (CeilF32, [DataValue::F32(a)]) => DataValue::F32(a.ceil()),
+                (CeilF64, [DataValue::F64(a)]) => DataValue::F64(a.ceil()),
+                (FloorF32, [DataValue::F32(a)]) => DataValue::F32(a.floor()),
+                (FloorF64, [DataValue::F64(a)]) => DataValue::F64(a.floor()),
+                (TruncF32, [DataValue::F32(a)]) => DataValue::F32(a.trunc()),
+                (TruncF64, [DataValue::F64(a)]) => DataValue::F64(a.trunc()),
+                _ => unreachable!(),
+            }])
+        });
```

Comment:
What do you think of defining this function in `function_generator.rs` next to the array of allowed libcalls?

Reply:
The only reason I'm hesitant to do this is that `fuzzgen` is supposed to be executor agnostic (i.e. icache needs none of this), so it feels kinda weird to have a interpreter specific function there.

Reply:
Ah, that makes sense. But this is weird for the same reason that the function references are: only outer fuzz target knows which functions and which libcalls are okay to call. So I think the set of allowed libcalls (and functions) should be provided by the fuzz target. But I also don't think that needs to block merging this PR, it's just something I think we should revisit soon.

Reply:
>So I think the set of allowed libcalls (and functions) should be provided by the fuzz target. 

I like that! 

> But I also don't think that needs to block merging this PR, it's just something I think we should revisit soon.

👍